gen/nvml/gpm.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nvml
+
+// nvml.GpmMetricsGet()
+type GpmMetricsGetVType struct {
```

Comment:
Where is this type actually used? Similarly for `GpmSupport*` below.

Reply:
Yeah. I wanted it at the top of the full set of functions related to `GpmMetricsGet()`. I could move the actual `GpmMetricsGet()` to the top though if you think that's clearer.

Reply:
It's the return value of a call to `GpmMetricsGetV()`.

The usage is:
```
var MetricsGet GpmMetricsGetType)
ret := GpmMetricsGetV(&MetricsGet).V1()
```

Reply:
I suppose my question is where do we expliclity need it, or is this something that a user needs to take care of. If that is the case we may want to add additional documentation to the affected functions.

Reply:
It is the pattern I have been following for these new "versioned" structs that the NVML team has been introducing. The API using the latest struct version is always available using e.g. `GpmMetricsGet(...)`, but if you want to explicitly use a different version you can call e.g. `GpmMetricsGetV(...).V1()`, `GpmMetricsGetV(...).V2(), ... instead.

I first started doing it with NVMl 11.6 here:
https://github.com/NVIDIA/go-nvml/pull/40/files#diff-0298839f64113aa3439ac75706f86ef591ba5c852ef6a740e30fb52322471488R1974

Reply:
OK. So a user *could* use these once we have multiple versions but it is not required here.